src/libraries/Common/src/System/IO/FileSystem.Attributes.Windows.cs

```diff
@@ -49,53 +60,64 @@ public static bool FileExists(string fullPath)
         /// <param name="returnErrorOnNotFound">Return the error code for not found errors?</param>
         internal static int FillAttributeInfo(string? path, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data, bool returnErrorOnNotFound)
         {
-            int errorCode = Interop.Errors.ERROR_SUCCESS;
-
             // Neither GetFileAttributes or FindFirstFile like trailing separators
             path = PathInternal.TrimEndingDirectorySeparator(path);
 
             using (DisableMediaInsertionPrompt.Create())
             {
-                if (!Interop.Kernel32.GetFileAttributesEx(path, Interop.Kernel32.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, ref data))
+                return FillAttributeInfoSlim(path, ref data, returnErrorOnNotFound);
+            }
+        }
+
+        /// <summary>
+        /// Same as <see cref="FillAttributeInfo"/> without separator-trimming and media-insertion blocking
+        /// </summary>
+        /// <param name="path">The file path from which the file attribute information will be filled.</param>
+        /// <param name="data">A struct that will contain the attribute information.</param>
+        /// <param name="returnErrorOnNotFound">Return the error code for not found errors?</param>
+        internal static int FillAttributeInfoSlim(string? path, ref Interop.Kernel32.WIN32_FILE_ATTRIBUTE_DATA data, bool returnErrorOnNotFound)
+        {
+            int errorCode = Interop.Errors.ERROR_SUCCESS;
+            string? prefixedString = PathInternal.EnsureExtendedPrefixIfNeeded(path);
+
+            if (!Interop.Kernel32.GetFileAttributesExPrivate(prefixedString, Interop.Kernel32.GET_FILEEX_INFO_LEVELS.GetFileExInfoStandard, ref data))
+            {
+                errorCode = Marshal.GetLastWin32Error();
+
+                Interop.Kernel32.WIN32_FIND_DATA findData = default;
+                if (!IsPathUnreachableError(errorCode))
                 {
-                    errorCode = Marshal.GetLastWin32Error();
+                    // Assert so we can track down other cases (if any) to add to our test suite
+                    Debug.Assert(errorCode is Interop.Errors.ERROR_ACCESS_DENIED or Interop.Errors.ERROR_SHARING_VIOLATION or Interop.Errors.ERROR_SEM_TIMEOUT,
```

Comment:
Please don't change style like this in the same PR -- we should generally leave existing code in the same style as other nearby code. If we need to change it, we would do that en-masse in its own PR.